tensornetwork/backends/symmetric/symmetric_backend.py

```diff
@@ -163,6 +163,141 @@ def conj(self, tensor: Tensor) -> Tensor:
+          "Use `reorthogonalize=True` for `numeig > 1`".format(numeig))
+    if initial_state is None:
+      if (shape is None) or (dtype is None):
+        raise ValueError("if no `initial_state` is passed, then `shape` and"
+                         "`dtype` have to be provided")
+      initial_state = self.randn(shape, dtype)
+
+    if not isinstance(initial_state, BlockSparseTensor):
+      raise TypeError("Expected a `BlockSparseTensor`. Got {}".format(
+          type(initial_state)))
+
+    vector_n = initial_state
+    vector_n.contiguous()# bring into contiguous memory layout
+
+    Z = self.norm(vector_n)
+    vector_n /= Z
+    norms_vector_n = []
+    diag_elements = []
+    krylov_vecs = []
+    first = True
+    eigvalsold = []
+    for it in range(num_krylov_vecs):
+      #normalize the current vector:
+      norm_vector_n = self.norm(vector_n)
+      if abs(norm_vector_n) < delta:
+        # we found an invariant subspace, time to stop
+        break
+      norms_vector_n.append(norm_vector_n)
+      vector_n = vector_n / norms_vector_n[-1]
+      #store the Lanczos vector for later
+      if reorthogonalize:
+        #vector_n is always in contiguous memory layout at this point
+        for v in krylov_vecs:
+          v.contiguous() # make sure storage layouts are matching
+          # it's save to operate on the tensor data now (pybass some checks)
+          vector_n.data -= numpy.dot(numpy.conj(v.data), vector_n.data) * v.data
+      krylov_vecs.append(vector_n)
+      A_vector_n = A(vector_n, *args)
+      A_vector_n.contiguous() #contiguous memory layout
+
+      # operate on tensor-data for scalar products
+      # this can be potentially problematic if vector_n and A_vector_n
+      # have non-matching shapes due to an erroneous matvec.
+      # If this is the case though an error will be thrown at line 281
+      diag_elements.append(
+          numpy.dot(numpy.conj(vector_n.data), A_vector_n.data))
+
+      if ((it > 0) and (it % ndiag) == 0) and (len(diag_elements) >= numeig):
+        #diagonalize the effective Hamiltonian
+        A_tridiag = numpy.diag(diag_elements) + numpy.diag(
+            norms_vector_n[1:], 1) + numpy.diag(
+                numpy.conj(norms_vector_n[1:]), -1)
+        eigvals, u = numpy.linalg.eigh(A_tridiag)
+        if not first:
+          if numpy.linalg.norm(eigvals[0:numeig] - eigvalsold[0:numeig]) < tol:
+            break
+        first = False
+        eigvalsold = eigvals[0:numeig]
+      if it > 0:
+        A_vector_n -= (krylov_vecs[-1] * diag_elements[-1])
+        A_vector_n -= (krylov_vecs[-2] * norms_vector_n[-1])
+      else:
+        A_vector_n -= (krylov_vecs[-1] * diag_elements[-1])
+      vector_n = A_vector_n
+
+    A_tridiag = numpy.diag(diag_elements) + numpy.diag(
+        norms_vector_n[1:], 1) + numpy.diag(numpy.conj(norms_vector_n[1:]), -1)
+    eigvals, u = numpy.linalg.eigh(A_tridiag)
+    eigenvectors = []
+    if numpy.iscomplexobj(A_tridiag):
```

Comment:
Wouldn't it make sense to always do this, regardless of whether it's complex or not?

Reply:
Yes, that's right. removed it